command/rootchain/fund/fund.go

```diff
@@ -96,7 +96,7 @@ func runCommand(cmd *cobra.Command, _ []string) {
 		rootTokenAddr types.Address
 	)
 
-	if params.mintRootToken {
+	if params.mintRootToken || params.deployerPrivateKey != "" {
```

Comment:
To me, it would be more correct to split this condition into two:
1. set `deployerKey` just in case non-empty string is provided for the deployer private key
2. set `rootTokenAddr` only if `mintRootToken` is set to true.

Reply:
Fixed through [33bab10](https://github.com/0xPolygon/polygon-edge/pull/1540/commits/33bab10d3e144ea3e3006fae3550dcf797506ba4)

Since in e2e we are providing --mint flag, but do not provide --private-key flag, this would not work. I agreed with Stefan that we should initialize deployerKey always, and rootTokenAddr only if --mint flag is set.